libs/feature/dataviz/src/lib/chart-view/chart-view.component.ts

```diff
@@ -129,10 +154,34 @@ export class ChartViewComponent {
       }
     })
   )
-  chartType: InputChartType = 'bar'
-  xProperty$ = new BehaviorSubject<string>(undefined)
-  yProperty$ = new BehaviorSubject<string>(undefined)
-  aggregation$ = new BehaviorSubject<FieldAggregation[0]>('sum')
+
+  get permalinkBaseUrl() {
+    return this.permalinkBaseUrlToken?.charAt(0) === '/'
+      ? `${window.location.origin}${this.permalinkBaseUrlToken}`
+      : this.permalinkBaseUrlToken
+  }
+
+  permalink$ = combineLatest([
+    this.xProperty$.pipe(filter((value) => value !== undefined)),
+    this.yProperty$.pipe(filter((value) => value !== undefined)),
+    this.aggregation$,
+    this.chartType$,
+  ]).pipe(
+    map(([xProp, yProp, aggregation, chartType]) => {
+      const link = `${this.permalinkBaseUrl}
+?e=gn-dataset-view-chart
+&a=api-url=${this.config.basePath}
+&a=aggregation=${aggregation}
+&a=x-axis=${xProp}
+&a=y-axis=${yProp}
+&a=chart-type=${chartType}
+&a=primary-color=%230f4395
+&a=secondary-color=%238bc832
+&a=main-color=%23555
+&a=background-color=%23fdfbff`
```

Comment:
I would use the `URL` class here to build the permalink, it's just safer

Reply:
Besides then you don't need the logic in `permalinkBaseUrl`.

Basically for me manipulating an url as a string is always a risk!

Reply:
Ah, good point, didn't think of `URL` handling absolute and relative urls